lcls_tools/emit_scan/mat_emit_scan.py

```diff
@@ -11,196 +14,247 @@
+            data = sio.loadmat(mat_file)['data'][0][0]
+            self._fields = data.dtype.names
+            self._file = mat_file
+            self._status = self._unpack_prop(STAT, data)
+            self._scan_type = self._unpack_prop(SCAN_TYPE, data)
+            self._name = self._unpack_prop(NAME, data)
+            self._quad_name = self._unpack_prop(QUAD_NAME, data)
+            self._quad_vals = self._unpack_prop(QUAD_VALS, data)
+            self._use = self._unpack_prop(USE, data)
+            self._ts = self._unpack_prop(TS, data)
+            self._beam = self._unpack_beam(data)  # data[7][iteration][fit][names]
+            self._charge = self._unpack_prop(Q, data)
+            self._charge_std = self._unpack_prop(Q_STD, data)
+            self._r_matrix = self._unpack_prop(R_MAT, data)  # List of r matrix per iteration
+            self._twiss_0 = self._unpack_prop(TWISS_0, data) #data[14]  # No clue how these fields are arranged
+            self._energy = self._unpack_prop(ENERGY, data)
+            self._twiss = self._unpack_prop(TWISS, data)
+            self._twiss_std = self._unpack_prop(TWISS_STD, data)
+            self._orbit = self._unpack_prop(ORBIT, data)
+            self._orbit_std = self._unpack_prop(ORBIT_STD, data)
+            self._twiss_pv = self._unpack_twiss_pv(data)
         except Exception as e:
-            print('error loading mat file: {0}'.format(e))
-
-    def _unpack_mat_data(self, mat_file):
-        data = sio.loadmat(mat_file)['data'][0][0]
-        self._file = mat_file
-        self._status = [status[0] for status in data[0]]
-        self._type = str(data[1][0])
-        self._name = str(data[2][0][0][0])
-        self._quad_name = str(data[3][0])
-        self._quad_val = data[4][0]
-        self._use = [use[0] for use in data[5]]
-        self._ts = data[6][0][0]
-        self._beam = self._unpack_beam(data[7], data[8])  # data[7][iteration][fit][names]
-        self._charge = data[11][0][0]
-        self._charge_std = data[12][0][0]
-        self._r_matrix = data[13][0]  # List of r matrix per iteration
-        self._twiss_0 = data[14]  # No clue how these fields are arranged
-        self._energy = data[15][0][0]  # GeV I believe
-        self._twiss = data[16]
-        self._twiss_std = data[17]
-        self._orbit = data[18]
-        self._orbit_std = data[19]
-        self._twiss_pv = self._unpack_twiss_pv(data[20])
-
-    @property
-    def file(self):
+            print('Error loading mat file: {0}'.format(e))
+
+    @property
+    def fields(self):
+        """The names of meta data fields associated with scan"""
+        return self._fields
+
+    @property
+    def mat_file(self):
+        """Mat file loaded"""
         return self._file
 
     @property
     def status(self):
-        return self._status
+        """Array of 0 or 1, no idea what status it's looking at"""
+        if self._status is not None:
+            return [status[0] for status in self._status]
```

Comment:
Some of the zero indexing here on out is scary again....but forced by Henrik.
Flagging for later.